clang/test/CodeGenCoroutines/coro-structured-concurrency.cpp

```diff
@@ -0,0 +1,84 @@
+// This file tests the coro_structured_concurrency attribute semantics. 
+// RUN: %clang_cc1 -std=c++20 -disable-llvm-passes -emit-llvm %s -o - | FileCheck %s
+
+#include "Inputs/coroutine.h"
+#include "Inputs/utility.h"
+
+template <typename T>
+struct [[clang::coro_structured_concurrency]] Task {
```

Comment:
Afaict, this particular `Task` can also be annotated using `coro_only_destroy_when_complete`?

I guess many libraries would want to use both `coro_structured_concurrency` / `coro_inplace_task` and `coro_only_destroy_when_complete`. Do those optimizations interact well with each other? Does it make sense to add a test case which tests both attributes together?

Reply:
Actually no. It's perfect valid to destroy this task before completion. See this as an example:
```
Task bar();
Task foo() {
   co_await bar();
}
```
You can totally call foo and destroy it, as long as `bar` is not running in parallel and you don't try to resume `bar` (through a handle you smuggled with the task, which this type does not allow to do)

Reply:
> through a handle you smuggled with the task, which this type does not allow to do

that's the point I am getting at. Most `task` types do not allow smuggling handles out of the wrappers types. As such, I expect that many task types would be annotated with both `coro_structured_concurrency` / `coro_inplace_task` and `coro_only_destroy_when_complete`. I guess we should have a test case that both attributes interact nicely with each other?